docs/configuration.md

```diff
@@ -456,6 +456,29 @@ Apart from these, the following properties are also available, and may be useful
     from JVM to Python worker for every task.
   </td>
 </tr>
+<tr>
+  <td><code>spark.jupyter.eagerEval.enabled</code></td>
```

Comment:
Should we add `sql`? E.g., `spark.sql.jupyter.eagerEval.enabled`? Because this is just for SQL `Dataset`. @HyukjinKwon @ueshin

Reply:
Oh, yes, I'd prefer to add `sql`.

Reply:
Got it, fix it in next commit.

Reply:
Is this Jupyter specific? Can we make it more general?

Reply:
btw the config flag isn't jupyter specific.

Reply:
Maybe rename to `spark.sql.repl.eagerEval.enabled` instead? It should also be documented that this is a hint. There's no guarantee that the repl you're using supports eager evaluation.

Reply:
Thanks, done. feb5f4a.